### PR TITLE
[Arm-199] fix: emergency API now can handle null emails

### LINKED ISSUE #20: Emergency API needs to handle null emails and phone numbers
[ARM-199](https://securityunion.atlassian.net/browse/ARM-199)

When changing state of myself from `Normal` -> `Emergency` with an account that is only signed in via phone number, and no email is associated with the account the emergency server panics because the email in postgres is null. The emergency server should be resilient to null emails and null phone numbers but at least one of them is guaranteed to exist due to a constraint placed on the database.

> Note: this change is working off of the [dev branch](https://github.com/security-union/armore-server/tree/dev).

```
[2021-01-30T17:27:22Z INFO  rocket::rocket] Matched: POST /v1/emergency/state application/json (update_state)
thread '<unnamed>' panicked at 'error retrieving column email: error deserializing column 1: a Postgres value was `NULL`', /rustc/d79f1bd31a1401b5d08096fcdf9a9eb23ddf95df/src/libstd/macros.rs:16:9
```

### PR DESCRIPTION
**Pull request by JasterV**
fixes #20

### CODE REVIEW

File: rust/src/controllers/emergency.rs [hunk 1]
```diff
@@ -128,15 +128,15 @@ fn build_emergency_notifications(
...
     state: &UserState,
-) -> (Vec<PushNotification>, Vec<Email>) {
+) -> (Vec<PushNotification>, Option<Email>) {
     get_user_details(rec_username, conn)
         .ok()
         .flatten()
-        .map_or((vec![], vec![]), |rec_details| {
+        .map_or((vec![], None), |rec_details| {
             let notifications =
                 build_emergency_notification(conn, &sender_details, &rec_details, &state);
```
Comment (griffobeid) [inline rust/src/controllers/emergency.rs:137]: Is this building SMS and push notifications?

Comment (JasterV) [reply rust/src/controllers/emergency.rs:137]: Its only building push notifications and emails for now, its not using the phone number to send SMS

Comment (JasterV) [reply rust/src/controllers/emergency.rs:137]: This is how it was in the node implementation when I made the translation

**Review by griffobeid — APPROVED** [review]

**Comment by griffobeid** [comment]
Thanks @JasterV bonus points for testing it :D

**Pull request merged**